pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -104,6 +105,26 @@ type VSphereConfig struct {
 	}
 }
 
+type Volumes interface {
+	// AttachDisk attaches given disk to given node. Current node
+	// is used when nodeName is empty string.
+	AttachDisk(vmDiskPath string, nodeName string) (diskID string, diskUUID string, err error)
+
+	// DetachDisk detaches given disk to given node. Current node
+	// is used when nodeName is empty string.
+	DetachDisk(volPath string, nodeName string) error
```

Comment:
Thanks for mentioning about this. I have made changes to methods DetachDisk() and DiskIsAttached() so it doesn't fail if node doesn't exist.